common/src/main/java/com/redhat/lightblue/rest/audit/LightblueAuditServletFilter.java

```diff
@@ -0,0 +1,527 @@
+
+
+            HttpServletResponse httpServletResponse = (HttpServletResponse) res;
+            HttpServletResponse httpServletResponseWrapperBuffered = new HttpServletResponseWrapperBuffered(httpServletResponse,new ByteArrayPrintWriter());
+            res = httpServletResponseWrapperBuffered;
+        }
+
+        if(auditReqFlag) {
+            stopwatch = Stopwatch.createStarted();
+        }
+
+        fChain.doFilter(hReq, res);
+
+        if(auditReqFlag) {
+            stopwatch.stop();
+            long elapsedTime = stopwatch.elapsed(TimeUnit.NANOSECONDS); // elapsedTime in ns
+            logEntryBuilder.setTimeElapsedInNs(elapsedTime);
+            HttpServletResponseWrapperBuffered httpServletResponseWrapperBuffered = (HttpServletResponseWrapperBuffered) res;
+            httpServletResponseWrapperBuffered.byteArrayPrintWriter.printWriter.flush();
+            byte[] bytes = httpServletResponseWrapperBuffered.byteArrayPrintWriter.toByteArray();
+            res.getOutputStream().write(bytes);
+            logEntryBuilder.setResponseSize(bytes.length);
+            final LogEntry logEntry = logEntryBuilder.createLogEntry();
+            String logEntryString = String.format(
+                    "Audited lightblue rest request => " +
+                            "{ " +
+                                "\"initialTimestamp\":\"%s\", " +
+                                "\"currentTimestamp\":\"%s\" , " +
+                                "\"principal\":\"%s\" , " +
+                                "\"resource\":\"%s\" , " +
+                                "\"operation\":\"%s\" , " +
+                                "\"entityName\":\"%s\" , " +
+                                "\"entityVersion\":\"%s\" , " +
+                                "\"entityStatus\":\"%s\" , " +
+                                "\"requestSize\":\"%d\" , " +
+                                "\"responseSize\":\"%d\" , " +
+                                "\"timeElapsedInNs\":\"%d\"  " +
+                            " }",
+                    logEntry.getTimestampText(),
+                    threadDateFormat.get().format(new Date()),
+                    logEntry.getPrincipal().getName(),
+                    logEntry.getResource(),
+                    logEntry.getOperation(),
+                    logEntry.getEntityName(),
+                    logEntry.getEntityVersion(),
+                    logEntry.getEntityStatus(),
+                    logEntry.getRequestSize(),
+                    logEntry.getResponseSize(),
+                    logEntry.getTimeElapsedInNs()
+            );
+            LOGGER.info(logEntryString);
+        }
+
+        if(LOGGER.isDebugEnabled()) {
+            LOGGER.debug("LightblueAuditServletFilter.doFilter invoked - end");
+        }
+    }
+
+    protected void setOperationEnittyVersionStatus(HttpServletRequest hReq, boolean isMetadata, LogEntryBuilder logEntryBuilder) {
```

Comment:
These method seems non-trivial. I wonder if we couldn't break it into smaller methods?

My other thought is maybe we create a single instance of Matcher, before the switch statement?  I am not typically a fan of re-using a variable, but maybe in this case it would help simplify? Then 'else if' statements could be used instead of embedded conditions.

Just thoughts, open for comments.